t/form_with_fields_passthrough_params.t

```diff
@@ -0,0 +1,246 @@
+#!perl -T
+
+use warnings;
+use strict;
+use Test::More 'no_plan';
+use Test::Fatal;
+use Test::Warnings ':all';
+use Test::Deep;
+use URI::file ();
+
+BEGIN {
+    delete @ENV{qw(PATH IFS CDPATH ENV BASH_ENV)};  # Placates taint-unsafe Cwd.pm in 5.6.1
+    use_ok( 'WWW::Mechanize' );
+}
+
+my $mech = WWW::Mechanize->new( cookie_jar => undef, autocheck => 0, strict_forms => 1, verbose_forms => 1 );
+isa_ok( $mech, 'WWW::Mechanize' );
+my $uri = URI::file->new_abs( 't/form_with_fields.html' )->as_string;
+
+$mech->get( $uri );
+ok( $mech->success, "Fetched $uri" ) or die q{Can't get test page};
+
+{
+    my $test = 'dies with no input';
+    like(
+        exception { my $form = $mech->form_with_fields(); },
+        qr/no fields provided/,
+        $test,
+    );
+}
+
+{
+    my $form;
+    cmp_deeply(
+        [ warnings { $form = $mech->form_with_fields(qw/1b/) } ],
+        [ re(qr/There are 2 forms with the named fields.  The first one was used./) ],
+        'warning on ambiguous match (1)',
+    );
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '1st_form', 'first form matches');
+}
+
+{
+    my $form = $mech->form_with_fields('1b', 'opt[2]');
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '2nd_form', 'second form matches');
+}
+
+{
+    my $form;
+    cmp_deeply(
+        [ warnings { $form = $mech->form_with_fields('4a', '4b') } ],
+        [ re(qr/There are 2 forms with the named fields.  The first one was used./) ],
+        'warning on ambiguous match (2)',
+    );
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '4th_form_1', 'fourth form matches');
+}
```

Comment:
This test seems very similar to the one two blocks above. Does it test different functionality?

Reply:
I copied all of those from the existing form_with_fields.t, because I wanted to make sure all the tests we already have also pass when strict is on. I did not actually prove those tests to be correct in this PR. I think the idea for these two is that in the `1st_form` test, it tests the outcome of calling `form_with_fields()` with one argument, while the `4th_form_1` test calls it with two arguments. I'd like to keep those in for consistency.